path_test.go

```diff
@@ -0,0 +1,42 @@
+package txtdirect
+
+import (
+	"testing"
+)
+
+func Test_zoneFromPath(t *testing.T) {
+	tests := []struct {
+		host     string
+		path     string
+		expected string
+		err      error
+	}{
+		{
+			"example.com",
+			"/caddy/v1",
+			"_redirect.v1.caddy.example.com",
+			nil,
+		},
+		{
+			"example.com",
+			"/1/2",
+			"_redirect.2.1.example.com",
+			nil,
+		},
+		{
+			"example.com",
+			"/",
+			"_redirect.example.com",
```

Comment:
"_redirect.example.com" would just lead to the path record. I think I suggested using "_redirect._.example.com", but that might not feel right.
Vague idea:
* if path == "/" + to != nil in `_redirect.example.com`-> rec.To
* if path == "/" + to == nil in `_redirect.example.com` -> rec.to from `redirect._.example.com

Reply:
Does that make sense? @alarsyo

Reply:
Mhh thinking about it. The general hierarchy of records with path in `example.com` would be:

specific zone www.example.com > wildcard zone _.example.com > default zone `to=` from .example.com

That would mean the above idea is incorrect as that would lead to the following hierarchy:
 default zone `to=` from .example.com > wildcard zone _.example.com

A few alternatives:
* use `default=` as the global path default in `example.com` <-- feels the most natural, would also be in line with the general host type I assume
* don't support an empty path redirect <-- probably the most common use case... so bad to not support it.
* do a workaround such as looking for _redirect._root.example.com <-- complex and unintuative

Reply:
I made some changes in my last commit to fix this. Now if path if "/", it returns host's TXT record and then redirects to ``default=``.